### PR TITLE
add usage and manpage for bmcdiscover "--sn" option

### PR DESCRIPTION
**Pull request by zet809**
For issue #3570 

Add usage and manpage for bmcdiscover "--sn" option

### CODE REVIEW

**Comment by zet809** [comment]
Hi, @whowutwut and @cxhong , will you pls take a look at this manpage and usage? Thx!

**Comment by zet809** [comment]
hi, @cxhong , will you pls review the manpage and usage of `bmcdiscover --sn`? Thx!

File: docs/source/guides/admin-guides/references/man1/bmcdiscover.1.rst [hunk 1]
```diff
@@ -56,6 +56,12 @@ OPTIONS
  
 
 
+\ **-**\ **-sn**\ 
+ 
+ Specify one or more service node on which bmcdiscover will run. In hierarchical cluster, the MN may not be able to access the BMC of CN directly, but SN can. With this option, bmcdiscover will be dispatched to the specified SN and set them as the 'servicenode' attribute of the discovered BMC node.
```
Comment (bybai) [inline docs/source/guides/admin-guides/references/man1/bmcdiscover.1.rst]: In the first sentence "one or more service node" -> "one or more service nodes".
I think this part means:
In hierarchical cluster, if the MN is not able to access the BMC of CN directly, with this option, bmcdiscover will be dispatched to the specified SN , then sets the SN as the 'servicenode' attribute of the discovered BMC node.

File: docs/source/guides/admin-guides/references/man1/bmcdiscover.1.rst [hunk 2]
```diff
@@ -142,31 +148,54 @@ EXAMPLES
...
+2. To get all BMCs in IP range "10.4.22-23.100-254", displayed in xCAT stanza format:
 
 
 .. code-block:: perl
 
      bmcdiscover -s nmap --range "10.4.22-23.100-254" -z
 
 
-3. Discover the BMCs and write the discovered-node definitions into the xCAT database and write out the stanza foramt to the console:
+3. To discovery BMCs through sn01:
```
Comment (bybai) [inline docs/source/guides/admin-guides/references/man1/bmcdiscover.1.rst]: To discovery --> To discover

File: xCAT-client/pods/man1/bmcdiscover.1.pod [hunk 3]
```diff
@@ -90,19 +94,34 @@ Display version information
 
 Note: Input for IP range can be in the form: scanme.nmap.org, microsoft.com/24, 192.168.0.1; 10.0.0-255.1-254.
 
-2. To get all BMSs in IP range "10.4.22-23.100-254", displayed in xCAT stanza format:
+2. To get all BMCs in IP range "10.4.22-23.100-254", displayed in xCAT stanza format:
 
     bmcdiscover -s nmap --range "10.4.22-23.100-254" -z
 
-3. Discover the BMCs and write the discovered-node definitions into the xCAT database and write out the stanza foramt to the console:
+3. To discovery BMCs through sn01:
```
Comment (bybai) [inline xCAT-client/pods/man1/bmcdiscover.1.pod]: To discovery --> To discover

File: xCAT-client/pods/man1/bmcdiscover.1.pod [hunk 4]
```diff
@@ -33,6 +33,10 @@ Note: The scan method currently support is B<nmap>.
 
 Specify one or more IP ranges acceptable to nmap.  IP range can be hostnames, IP addresses, networks, etc.  A single IP address (10.1.2.3) or an IP range (10.1.2.0/24) can be specified.  If the range is very large, the B<bmcdiscover> command may take a long time to return. 
 
+=item B<--sn>
+
+Specify one or more service node on which bmcdiscover will run. In hierarchical cluster, the MN may not be able to access the BMC of CN directly, but SN can. With this option, bmcdiscover will be dispatched to the specified SN and set them as the 'servicenode' attribute of the discovered BMC node.
```
Comment (bybai) [inline xCAT-client/pods/man1/bmcdiscover.1.pod]: one or more service node --> one or more service nodes

**Comment by bybai** [comment]
I am OK to merge it.

**Pull request merged**